content/references/protocol-reference/transactions/transaction-types/ammdeposit.md

```diff
@@ -0,0 +1,131 @@
+
+```json
+{
+    "Account" : "rJVUeRqDFNs2xqA7ncVE6ZoAhPUoaJJSQm",
+    "Amount" : {
+        "currency" : "TST",
+        "issuer" : "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd",
+        "value" : "2.5"
+    },
+    "Amount2" : "30000000",
+    "Asset" : {
+        "currency" : "TST",
+        "issuer" : "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd"
+    },
+    "Asset2" : {
+        "currency" : "XRP"
+    },
+    "Fee" : "10",
+    "Flags" : 1048576,
+    "Sequence" : 7,
+    "TransactionType" : "AMMDeposit"
+}
+```
+
+{% include '_snippets/tx-fields-intro.md' %}
+
+| Field         | JSON Type           | [Internal Type][] | Required? | Description |
+|:--------------|:--------------------|:------------------|:----------|:------------|
+| `Asset`       | Object              | STIssue           | Yes       | The definition for one of the assets in the AMM's pool. In JSON, this is an object with `currency` and `issuer` fields (omit `issuer` for XRP). |
+| `Asset2`      | Object              | STIssue           | Yes       | The definition for the other asset in the AMM's pool. In JSON, this is an object with `currency` and `issuer` fields (omit `issuer` for XRP). |
+| `Amount`      | [Currency Amount][] | Amount            | No        | The amount of one asset to deposit to the AMM. If present, this must match the type of one of the assets (tokens or XRP) in the AMM's pool. |
+| `Amount2`     | [Currency Amount][] | Amount            | No        | The amount of another asset to add to the AMM. If present, this must match the type of the other asset in the AMM's pool and cannot be the same asset as `Amount`. |
+| `EPrice`      | [Currency Amount][] | Amount            | No        | The maximum effective price, in the deposit asset, to pay for each LP Token received. |
+| `LPTokenOut`  | [Currency Amount][] | Amount            | No        | How many of the AMM's LP Tokens to buy. |
+
+
+### AMMDeposit Modes
+
+This transaction has several modes, depending on which flags you specify. Each mode expects a specific combination of fields. The modes fall into two categories: 
+
+- **Double-asset deposits**, in which you provide both assets in the AMM's pool in proportions that match the balance of those assets already there. These deposits are not subject to a fee.
+- **Single-asset deposits**, in which you provide only one of the AMM's two assets. You can envision a single-asset deposit as trading _some_ of the deposited asset for the other asset, then performing a double-asset deposit. The trading fee applies to the amount you would need to trade for, but not to the rest of the deposit; the amount of the fee is debited from the LP Tokens paid out. _For example, if the AMM's asset pool is split perfectly evenly between USD and EUR, and you try to deposit 100 USD, the amount of LP Tokens you receive is slightly less than if you had deposited 50 EUR + 50 USD, because you pay the trading fee to convert some of your USD to an equal amount of EUR._
```

Comment:
The way this is worded is confusing. I don't think we should use the example of trading one asset for another, since it implies you're getting a withdrawal of the other asset. Maybe we elaborate on how the AMM will try to balance the difference, and in doing so subract trading fees from LP token payouts as it tries to balance the values.

Reply:
The way this _works_ is confusing; this wording is at least less confusing to me than anything else I've come up with. 

I'm not sure what you mean by elaborating on "how the AMM will try to balance the difference" instead. The AMM doesn't move anything around to try and make things balance, it just charges the trading fee on a percentage of your deposit. And it was helpful to me at least to know that it's calculated so that the result is identical to if you did a trade against the AMM first and then did a double-asset deposit.

Reply:
Changed to this:

> - **Single-asset deposits**, in which you provide only one of the AMM's two assets. The AMM charges a fee, debited from the LP Tokens paid out, based on how much your deposit shifts the balance of assets in the pool.

Then I moved some details to the Single Asset Deposit Fee Calculations section below:

> ### Single Asset Deposit Fee Calculations
>
> The fee for a single asset deposit is calculated to be the same as if you had used the AMM to trade part of the deposit amount for the other asset, then performed a double-asset deposit. The AMM's trading fee applies to the amount you would need to trade for, but not to the rest of the deposit. _For example, if the AMM's asset pool is split perfectly evenly between USD and EUR, and you try to deposit 100 USD, the amount of LP Tokens you receive is slightly less than if you had deposited 50 EUR + 50 USD, because you pay the trading fee to convert some of your USD to an equal amount of EUR._